server/src/userManagement.mjs

```diff
@@ -147,6 +147,17 @@ function associateUserWithWs(userId, ws) {
   }
 }
 
+// Delete websocket object associated with a userId when that websocket connection is closed by a close event
+function deleteWsOfUser(ws, userId) {
+  let wsArray = [];
+  if (user2ws.has('' + userId)) {
+    wsArray = user2ws.get('' + userId)
+    wsArray = wsArray.filter((socketObject) => socketObject !== ws);
+    user2ws.set('' + userId, wsArray);
+  } else {
+    logger.warning(`userId ${userId} does not have associated websocket mapping`)
+  }
+}
```

Comment:
I recommend that we write a unit test for this new function

Reply:
Sure, added a corresponding unit test